File: rust/src/models/one_time_plan_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// One-time fee
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OneTimePlanFee {
    pub quantity: i32,

    pub unit_price: rust_decimal::Decimal,
}

impl OneTimePlanFee {
    pub fn new(quantity: i32, unit_price: rust_decimal::Decimal) -> Self {
        Self {
            quantity,
            unit_price,
        }
    }
}

File: typescript/src/models/oneTimePlanFee.ts
// this file is @generated
/** One-time fee */
export interface OneTimePlanFee {
  quantity: number;

  unitPrice: string;
}

export const OneTimePlanFeeSerializer = {
  _fromJsonObject(object: any): OneTimePlanFee {
    return {
      quantity: object["quantity"],
      unitPrice: object["unit_price"],
    };
  },

  _toJsonObject(self: OneTimePlanFee): any {
    return {
      quantity: self.quantity,
      unit_price: self.unitPrice,
    };
  },
};
